### ISSUE TITLE
[CNCF TOC] Archive Pravega Project

### ISSUE THREAD

**Issue by angellk** [pravega/pravega#7436]
It looks like work on the Pravega project decreased considerably in March, 2024.

Please confirm Pravega's current status by October 25, 2024. At that time the project will be moved to a vote to archive. If active development resumes on the project, the Pravega project may resubmit to apply for CNCF Sandbox.

xref: https://github.com/cncf/toc/issues/1460

**Comment by RaulGracia** [comment]
Hi @angellk, thanks for approaching the project.

We acknowledge the decay in  contributions in the last months. The reason for this have been changes in the main contributing organization supporting this project. Still, the project is being used and active in several fronts:

1.	There are 2 EU Projects (totaling 7.4M€) in which Pravega is used as the main streaming service (https://neardata.eu/, https://cloudskin.eu/). We wrote a blog post about this: https://cncf.pravega.io/blog/2023/06/06/pravega-in-european-research-projects/
2.	The Zookeeper Operator keeps getting open-source contributions and we are planning on expanding the
committer basis: https://github.com/pravega/zookeeper-operator/issues/617
3.	We are exploiting Pravega in several scientific publications showcasing the relevance of its unique features: https://dl.acm.org/doi/abs/10.1145/3590140.3629113, https://dl.acm.org/doi/abs/10.1145/3626562.3626828.
4. We are working with stakeholders ([National Center for Tumor Diseases, Germany](https://www.dkfz.de/en/nct/index.html)) exploiting Pravega for video analytics in computer-assisted surgery use cases: https://gitlab.com/nct_tso_public/cloudskin/nct_dell_demo 


Due to all that ongoing activity, probably more related to exploitation than development, I would ask you to give us more time to see if we can accumulate more critical mass in terms of contributions and keep the project updated. Otherwise, archiving right away it may have a negative impact on all these ongoing activities.

Thanks a lot,
Raúl.

**Comment by angellk** [comment]
Thank you for the update @RaulGracia - could you please engage with [TAG Contributor Strategy](https://github.com/cncf/tag-contributor-strategy) for guidance on growing your maintainers and please check back in with an update in 3 months.

**Comment by RaulGracia** [comment]
@angellk thanks, we will try our best to increase contributions and engagement within this 3-months period and even further. Thanks!

**Comment by RaulGracia** [comment]
FYI: https://github.com/cncf/tag-contributor-strategy/issues/747

**Comment by RaulGracia** [comment]
I'm summarizing the progress so far on the expansion of the Pravega open-source community during the last weeks, with the ultimate goal to keep a healthy number of contributions to the repository:
1. Pravega is a core technology of two EU projects (https://neardata.eu/, https://cloudskin.eu/). In this context, we are working with health-related institutions building PoCs that solve data management problems thank to Pravega tiered data streams. In addition to ([National Center for Tumor Diseases, Germany](https://www.dkfz.de/en/nct/index.html)) PoC we built and we are extending, we are now working with [UKHSA](https://www.gov.uk/government/organisations/uk-health-security-agency) as well for building a Pravega-based PoC for managing and processing genomic data. We have showcased this work in conferences like [HiPEAC'25](https://www.hipeac.net/2025/barcelona/#/program/sessions/8189/).
2. We have used Pravega in two new research publications: https://dl.acm.org/doi/abs/10.1145/3700824.3701099 https://dl.acm.org/doi/abs/10.1145/3700824.3701097. Hopefully, this will attract some attention from the scientific community.
3. We have added two new committers outside the main original sponsoring company to the project (@cheleb, @gfinol) that have demonstrated interest and willingness to contribute. Hopefully, they will help in bringing back more activity to the repo.
4. We have contacted TAG Contributor Strategy via an issue to get feedback on growing our community: https://github.com/cncf/tag-contributor-strategy/issues/747

In summary, we are working towards making the project self-sustainable. To this end, I would appreciate if you could give us i) more time to see the outcomes of some of these activities, and ii) specific CNCF resources/guidance to help us in this task.

Thanks,
Raúl.

**Comment by chira001** [comment]
Hi @RaulGracia - I'm from the CNCF TOC and checking in to understand progress.    The work on the project seems to have stalled - could you provide an update on status please?

**Comment by RaulGracia** [comment]
Hi @chira001, 

Building on our previous update, we’d like to share that while core development on Pravega has slowed, significant activity is ongoing through its integration into several EU-funded research initiatives. These efforts are focused on expanding the ecosystem and demonstrating Pravega’s capabilities in real-world, high-impact scenarios. A significant part of of the development effort around Pravega is being generated as part of the ecosystem of tools for specific use cases related 2 EU Research projects. You can check out the repositories: https://github.com/neardata-eu https://github.com/cloudskin-eu. Concrete examples of what we have done so far and what we are currently developing are:
- [Real-time surgery video analytics PoC with the National Center for Tumor Diseases (Germany)](https://github.com/neardata-eu/nct_dell_demo_pravega) 
- [Pravega video ingestion with embeddings generations from video frames for a surgiomics use case (National Center for Tumor Diseases, Germany)](https://github.com/neardata-eu/video-stream-indexing).
- Framework to use Pravega for building serverless data-intensive FaaS pipelines (_will be open sourced upon publication acceptance, due to review anonymity reasons_). We can show you the results in private meanwhile, if needed.
- We are working with UK Health Security Agency on a [PoC for executing streaming genomics pipelines on top of Pravega](https://github.com/neardata-eu/streaming-genomics) (repo not yet open).
- [Orchestrator to coordinate the dynamic parallelism of Pravega streams with Flink task managers](https://github.com/pravega/pravega-samples/tree/master/scenarios/pravega-flink-autoscaling) 
- [Integration of Pravega with an ephemeral storage system](https://github.com/cloudskin-eu/pravega-geds)
- We have also done efforts integrating and evaluating the Pravega clients running in [Scone confidential container technology](https://scontain.com/).
- Other benchmarks: https://github.com/neardata-eu/video-streaming-benchmarks https://github.com/neardata-eu/pravega-rust-benchmark

Given that Pravega already provides a robust core, our current focus is on ecosystem development and proof-of-concept implementations. These efforts aim to validate Pravega’s value in diverse domains, which could pave the way for sustainable development if adoption grows among partners. We plan to continue in this line through the duration of the EU projects (end of 2025 to early 2026) and will keep the community updated on major milestones.

Finally, we’ve also reached out to TAG Contributor Strategy to [get feedback on growing our community](https://github.com/cncf/tag-contributor-strategy/issues/747). We’re very open to support from the CNCF and welcome collaboration, contributions, and ideas from anyone interested in helping shape the future of Pravega. We’re actively looking to grow the community and would be glad to connect with potential contributors or maintainers.

Thanks a lot,
Raúl.
